services/memory_storage/memory_storage.cpp

```diff
@@ -256,19 +274,82 @@ namespace services {
                   session.data());
             return;
         }
+        auto collection_names = logical_plan->collection_dependencies();
+        context_storage_t collection_dependencies;
+        for (const auto& name : collection_names) {
+            if (!check_collection_(session, name)) {
+                trace(log_,
+                      "memory_storage_t:execute_plan_ collection not found {}, sesion: {}",
+                      name.to_string(),
+                      session.data());
+                return;
+            }
+            collection_dependencies.emplace(name, collections_.at(name).context);
```

Comment:
Do we need `name` later? may be we can move it?

Reply:
it is returned in a set, and there be structure preserving things happening...
i changed set to unordered set, now should be fine